### PR TITLE
update to 1.78.1

### PR DESCRIPTION
**Pull request by cbouss**
Changes:
- update to 1.78.1
- add patch for 3.12
- workaround test lib location libregress
- remove gettext - not directly needed and provided by libglib anyway.

https://gitlab.gnome.org/GNOME/gobject-introspection/-/tree/1.78.1?ref_type=tags

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -48,17 +49,20 @@ requirements:
...
+    - libffi 3.4
+    - pthread-stubs 0.3
+    - zlib {{ zlib }} # [win]
+    - setuptools
   run:
-    - libffi >=3.4,<3.5
     - python
-    - gettext  # [osx]
+    # through run_exports
+    - libffi >=3.4,<3.5
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]: `libffi` exports `>=3.4,<4.0a0`. Any reason to use a stricter constraint?

Comment (cbouss) [reply recipe/meta.yaml]: It does export x.x: https://github.com/AnacondaRecipes/libffi-feedstock/blob/master/recipe/meta.yaml#L24

Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml]: https://conda-metadata-app.streamlit.app/?q=pkgs%2Fmain%2Flinux-64%2Flibffi-3.4.4-h6a678d5_0.conda#run-exports says
```
weak:
- libffi >=3.4,<4.0a0
```
and when we look at https://staging.continuum.io/package/prefect/fs/gobject-introspection-feedstock/pr11/82039b7/linux-64/gobject-introspection-1.78.1-py310hbb6d50b_0.conda (the produced package metadata, I see `>=3.4,<4.0a0` is being injected.

**Review by JeanChristopheMorinPerso — APPROVED** [review]

File: recipe/meta.yaml (hunk 1 above)
Comment (cbouss) [reply recipe/meta.yaml]: ok `pin_subpackage` max_pin default is `x` and previous libffi pin was overly strict.
Thank you!

File: recipe/meta.yaml [hunk 2]
```diff
@@ -33,9 +32,11 @@ build:
 requirements:
   build:
     - python
+    # run-dep as a workaround for now; upstream package still imports from distutils
+    - setuptools  # [py>311]
     - {{ posix }}bison
     - {{ posix }}flex
-    - meson >=0.58.2
+    - meson >=1.2.0
```
Comment (Jrice1317) [inline recipe/meta.yaml]:
```suggestion
    - meson >=0.60.0
```
I see this [here](https://gitlab.gnome.org/GNOME/gobject-introspection/-/blob/1.78.1/meson.build?ref_type=tags#L3)

**Review by skupr-anaconda — APPROVED** [review]
LGTM but Jaida's comment

File: recipe/meta.yaml (hunk 2 above)
Comment (cbouss) [reply recipe/meta.yaml]: Thank you

File: recipe/bld.bat [hunk 3]
```diff
@@ -32,9 +32,9 @@ if errorlevel 1 exit 1
 ninja -v -C builddir -j %CPU_COUNT%
```
Comment (skupr-anaconda) [inline recipe/bld.bat:32]: Seems that `meson configure builddir` failed on win64:
`(%BUILD_PREFIX%) %SRC_DIR%>if errorlevel 1 exit 1`
Or the error is something else

Comment (cbouss) [reply recipe/bld.bat:32]: It was failing at the build step, but it seemed to have been a fluke.

**Pull request merged**